local-macros/src/lib.rs

```diff
@@ -32,29 +34,30 @@ impl AstTypeArgs {
     fn should_impl_from(&self) -> bool {
         self.impl_from.unwrap_or(true)
     }
-}
 
-#[proc_macro_attribute]
-pub fn ast_type(attr: TokenStream, item: TokenStream) -> TokenStream {
-    let item_for_parsing = item.clone();
-    let DeriveInput {
-        ident: ast_type_name,
-        data,
-        ..
-    } = parse_macro_input!(item_for_parsing);
-
-    let node_interface_and_readonly_text_range_implementation = match data {
-        Struct(struct_) => {
-            let first_field_name = match struct_.fields {
-                Fields::Named(FieldsNamed { named, .. }) => named
-                    .iter()
-                    .nth(0)
-                    .expect("Expected at least one struct field")
-                    .ident
-                    .clone(),
-                _ => panic!("Expected named fields"),
-            };
+    fn interfaces_vec(&self) -> Vec<String> {
+        let mut vec = vec!["ReadonlyTextRange".to_string(), "NodeInterface".to_string()];
+        vec.append(&mut self.interfaces.as_ref().map_or_else(
+            || vec![],
```

Comment:
Since appending an empty vec is a no-op, I'd find this a bit more clear if it were written as an `if let Some(...)` rather than `map_or_else`.

Reply:
Sure, updated to use `if let Some(...)` instead